beacon-chain/blockchain/kzg/kzg.go

```diff
@@ -2,14 +2,24 @@ package kzg
 
 import (
 	"errors"
+	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
```

Comment:
This shim file was intentionally made to not rely on other parts of the code --  Can you define NumOfColumns in this file as an alias for CellsPerExtBlob?

The idea is that this file and code by itself can be tested and checked to be correct in isolation to the rest of the code

Reply:
If the length only exists in that package, I don't think it makes sense to have it at all. I've pushed a commit which changes `CellsAndProofs` to use slices instead of arrays. Open to changing it though.

Reply:
I think it ideally gets checked by the caller who knows `fieldparams.NumOfColumns` --  There should probably be a check/test that breaks if `ckzg.CellsPerExtBlob != fieldparams.NumOfColumns`, since these values come from two different sources but should be the same.

If c-kzg for example, were to return less cells than the client code is expecting due to the params, then this would be caught by that test.

RE changing to slice, imo ideally this PR just updates the dependency and other changes which are not necessary are done in a separate PR.

Reply:
I don't like making this change in this PR either. But it was using a constant which no longer exists. `ckzg.CellsPerExtBlob` isn't available anymore, it's been made private.

Reply:
Oh I didn't realise that CellsPerExtBlob was now private from ckzg -- whether that should be private is out of scope of this PR, so this change makes sense in that context.